### DISCUSSION TITLE
math wallet rainbow wallet

### DISCUSSION THREAD

**Discussion by ryuk8000** [WalletConnect/walletconnect-monorepo#2405] [Developer Support]
math wallet isn't able to scan the qr code.
rainbow wallet is getting scanned but rejecting connection

**Comment by finessevanes** [comment]
Can you give us a little bit more information? Please include the dapp you're trying to connect to, the device you're using, the SDK version numbers.

**Reply by ryuk8000** [reply]
import { EthereumClient, w3mConnectors, w3mProvider } from '@web3modal/ethereum'
import { Web3Modal } from '@web3modal/html'
import { configureChains, createConfig } from '@wagmi/core'
import { arbitrum, mainnet, polygon } from '@wagmi/core/chains'

const chains = [arbitrum, mainnet, polygon]
const projectId = 'YOUR_PROJECT_ID'

const { publicClient } = configureChains(chains, [w3mProvider({ projectId })])
const wagmiConfig = createConfig({
  autoConnect: true,
  connectors: w3mConnectors({ projectId, version: 1, chains }),
  publicClient
})
const ethereumClient = new EthereumClient(wagmiConfig, chains)
const web3modal = new Web3Modal({ projectId }, ethereumClient)   using this not able to scan math wallet qr using mobile wallet of math wallet

**Reply by ryuk8000** [reply]
i am scanning qr of math wallet in desktop through a math wallet mobile app

**Reply by finessevanes** [reply]
Could you include a screenshot of what you're running into and the error. Apologies, but I'm not quite following where you're running into the error.

**Reply by ryuk8000** [reply]
the problem is solved for math wallet in dark theme of web3modal scanning of math  wallet is not working

**Reply by ryuk8000** [reply]
math wallet qr scan is working in light theme
